internal/deploy/docker.go

```diff
@@ -92,13 +112,61 @@ func (c *dockerDeploymentManifest) Inspect(service string) (*ServiceManifest, er
 		ID:         inspect.ID,
 		Name:       strings.TrimPrefix(inspect.Name, "/"),
 		Connection: service,
-		Hostname:   inspect.NetworkSettings.Networks["fleet_default"].Aliases[0],
+		Alias:      inspect.NetworkSettings.Networks["fleet_default"].Aliases[0],
+		Hostname:   inspect.Config.Hostname,
+		Platform:   inspect.Platform,
 	}, nil
 }
 
+// Mount will mount a service with ability to perform actions within that services environment
+// TODO: Not a fan of passing in installType here, should think about abstracting that portion out more
+func (c *dockerDeploymentManifest) Mount(service string, installType string) (installer.Package, error) {
+	log.WithFields(log.Fields{
+		"service":     service,
+		"installType": installType,
+	}).Trace("Mounting service for configuration")
+
+	container, _ := c.Inspect(service)
+	var install installer.Package
+	if strings.EqualFold(service, "elastic-agent") {
+		switch installType {
+		case "tar":
+			install = installer.NewElasticAgentTARPackage(container.Name, c.ExecIn, c.AddFiles)
+			return install, nil
+		case "rpm":
+			install = installer.NewElasticAgentRPMPackage(container.Name, c.ExecIn, c.AddFiles)
+			return install, nil
+		case "deb":
+			install = installer.NewElasticAgentDEBPackage(container.Name, c.ExecIn, c.AddFiles)
+			return install, nil
+		}
+	}
+
+	return nil, nil
+}
+
 // Remove remove services from deployment
 func (c *dockerDeploymentManifest) Remove(services []string, env map[string]string) error {
-	serviceManager := compose.NewServiceManager()
+	for _, service := range services[1:] {
+		manifest, _ := c.Inspect(service)
+		_, err := shell.Execute(c.Context, ".", "docker", "rm", "-f", manifest.Name)
```

Comment:
Eventually remove volumes too?
```suggestion
		_, err := shell.Execute(c.Context, ".", "docker", "rm", "-fv", manifest.Name)
```

if the RemoveServicesFromCompose is not going to be used, I think it could be deleted.